noxfile.py

```diff
@@ -100,53 +123,62 @@ def test_sphinx(session: nox.Session, sphinx: int) -> None:
     session.run("pytest", *session.posargs)
 
 
-@nox.session()
+@session()
 def translate(session: nox.Session) -> None:
-    """Translation commands. Available commands after `--` : extract, update, compile, init."""
-    # get the command from posargs, default to "update"
-    pybabel_cmd, found = ("update", False)
-    for c in ["extract", "update", "compile", "init"]:
-        if c in session.posargs:
-            pybabel_cmd, found = (c, True)
-
-    if found is False:
-        print(
-            "No translate command found. Use like: `nox -s translate -- COMMAND`."
-            "\ndefaulting to `update`"
-            "\nAvailable commands: extract, update, compile, init"
-        )
-
-    # get the language from parameters default to en.
-    # it can be deceiving but we don't have a table of accepted languages yet
-    lan = "en" if len(session.posargs) < 2 else session.posargs[-1]
-
-    # get the path to the differnet local related pieces
-    locale_dir = str(ROOT / "src" / "pydata_sphinx_theme" / "locale")
-    babel_cfg = str(ROOT / "babel.cfg")
-    pot_file = str(locale_dir / "sphinx.pot")
-
-    # install deps
+    """Update translation related files."""
+    session.install("Babel", "jinja2")
+
+    # fmt: off
+    session.run(  # generate/update the .pot file
+        "pybabel", "extract", ".",
+        "-F", str(babel_cfg.relative_to(root_dir)),
+        "-o", str(pot_file.relative_to(root_dir)),
+        #"-k", "'_ __ l_ lazy_gettext'",
+    )
+    # fmt: on
+
+    # update the message catalog (.po)
+    languages = [f.stem for f in locale_dir.glob("*") if f.is_dir()]
```

Comment:
```suggestion
    languages = [f.stem for f in locale_dir.iterdir() if f.is_dir()]
```
should be more efficient, avoids unnecessary pattern matching against `*`